fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,290 @@
+                                                                    stub_name=_grpc_rpc_stub_method_name)
+                    if grpc_function_json_data:
+                        if stub_method[1] is not None:
+                            if sys.version[0] == '2':
+                                # convert unicode strings to utf-8 for python 2.7
+                                import ast
+                                grpc_function_json_data_tmp = ast.literal_eval(json.dumps(grpc_function_json_data))
+                            else:
+                                grpc_function_json_data_tmp = grpc_function_json_data
+
+                            response_received_tmp = stub_method[1](grpc_request_tuple[1](**grpc_function_json_data_tmp))
+
+                            # Closing the gRPC Server Connection
+                            grpc_channel_obj.close()
+                        else:
+                            raise FunctionError("Stub method not found, please specify the valid stub method.")
+                    else:
+                        raise FunctionError("Please Specify the valid data input form the Resilient.")
+                except grpc.RpcError as rpc_err:
+                    status_code = rpc_err.code()
+                    log.debug(
+                        "{}\n connection status :{} \n name : {}\n value : {}".format(rpc_err, rpc_err.details(),
+                                                                                      status_code.name,
+                                                                                      status_code.value))
+                    raise FunctionError("Server Connection status : {}".format(rpc_err.details()))
+                else:
+                    log.info("Response received from the server : {}".format(response_received_tmp))
+            else:
+                raise NotImplementedError(
+                    "Not Implemented these {} communication types ".format(_grpc_communication_type))
+
+            # Attempting to convert received data into JSON object
+            try:
+                if isinstance(response_received_tmp, dict):
+                    response_received = response_received_tmp
+                else:
+                    response_received = json.loads(response_received_tmp)
+                log.debug("The Received Data is Converted into JSON Object : {}".format(response_received))
+            except Exception as e:
+                try:
+                    response_received = MessageToDict(response_received_tmp)
+                except Exception as e:
+                    log.debug("The Received data is not converted into JSON Object {}".format(e))
+                    response_received = str(response_received_tmp)
+
+            yield StatusMessage("done...")
+
+            results = {
+                "content": response_received,
+                "channel": grpc_channel
+            }
+            # Produce a FunctionResult with the results
```

Comment:
End the function similarly to: https://github.com/neetinkandhare/resilient-community-apps/blob/fn_alien_vault/fn_alienvault_otx/fn_alienvault_otx/components/fn_alienvault_otx_threat_lookup.py#L102

Reply:
ok